ydb/core/tx/schemeshard/schemeshard_export_scheme_uploader.cpp

```diff
@@ -0,0 +1,300 @@
+#include <ydb/core/wrappers/s3_storage_config.h>
+#include <ydb/core/wrappers/s3_wrapper.h>
+#include <ydb/library/actors/core/actor_bootstrapped.h>
+#include <ydb/library/actors/core/hfunc.h>
+#include <ydb/public/api/protos/ydb_export.pb.h>
+#include <ydb/public/lib/ydb_cli/dump/util/view_utils.h>
+
+namespace NKikimr::NSchemeShard {
+
+class TSchemeUploader: public TActorBootstrapped<TSchemeUploader> {
+
+    using TS3ExternalStorageConfig = NWrappers::NExternalStorage::TS3ExternalStorageConfig;
+    using TEvExternalStorage = NWrappers::TEvExternalStorage;
+    using TPutObjectResult = Aws::Utils::Outcome<Aws::S3::Model::PutObjectResult, Aws::S3::S3Error>;
+
+    static TString BuildViewScheme(const TString& path, const NKikimrSchemeOp::TViewDescription& viewDescription, const TString& backupRoot, TString& error) {
+        NYql::TIssues issues;
+        auto scheme = NYdb::NDump::BuildCreateViewQuery(viewDescription.GetName(), path, viewDescription.GetQueryText(), backupRoot, issues);
+        if (!scheme) {
+            error = issues.ToString();
+        }
+        return scheme;
+    }
+
+    bool BuildSchemeToUpload(const NKikimrScheme::TEvDescribeSchemeResult& describeResult, TString& error) {
+        const auto pathType = describeResult.GetPathDescription().GetSelf().GetPathType();
+        switch (pathType) {
+            case NKikimrSchemeOp::EPathTypeView: {
+                Scheme = BuildViewScheme(describeResult.GetPath(), describeResult.GetPathDescription().GetViewDescription(), DatabaseRoot, error);
+                return !Scheme.empty();
+            }
+            default:
+                error = TStringBuilder() << "unsupported path type: " << pathType;
+                return false;
+        }
+    }
+
+    void HandleSchemeDescription(TEvSchemeShard::TEvDescribeSchemeResult::TPtr& ev) {
+        const auto& describeResult = ev->Get()->GetRecord();
+
+        LOG_D("HandleSchemeDescription TEvSchemeShard::TEvDescribeSchemeResult"
+            << ", self: " << this->SelfId()
+            << ", status: " << describeResult.GetStatus()
+        );
+
+        auto reportError = [&](const TString& error) {
+            Send(SchemeShard, new TEvPrivate::TEvExportSchemeUploadResult(
+                ExportId, ItemIdx, false, error
+            ));
+        };
+
+        if (describeResult.GetStatus() != TEvSchemeShard::EStatus::StatusSuccess) {
+            reportError(describeResult.GetReason());
+            return;
+        }
+
+        TString error;
+        if (!BuildSchemeToUpload(describeResult, error)) {
+            reportError(error);
+            return;
+        }
+
+        if (auto permissions = NDataShard::GenYdbPermissions(describeResult.GetPathDescription()); !permissions.Defined()) {
+            reportError("cannot infer permissions");
+            return;
+        } else {
+            google::protobuf::TextFormat::PrintToString(permissions.GetRef(), &Permissions);
+        }
```

Comment:
Странно видеть обращение к `permissions` в `else`-ветке. Скорее, логично написать как-то так:
```suggestion
        if (auto permissions = NDataShard::GenYdbPermissions(describeResult.GetPathDescription())) {
            google::protobuf::TextFormat::PrintToString(permissions.GetRef(), &Permissions);
        } else {
            reportError("cannot infer permissions");
            return;
        }
```